src/interpolations.jl

```diff
@@ -915,6 +920,37 @@ function value(ip::Lagrange{3,RefPrism,2}, i::Int, ξ::Vec{3})
     throw(ArgumentError("no shape function $i for interpolation $ip"))
 end
 
+
+#####################################
+# Lagrange dim 3 RefPyramid order 1 #
+#####################################
+# Build on https://defelement.com/elements/examples/pyramid-Lagrange-1.html
+
+getnbasefunctions(::Lagrange{3,RefPyramid,1}) = 5
+
+vertexdof_indices(::Lagrange{3,RefPyramid,1}) = ((1,), (2,), (3,), (4,), (5,),)
+facedof_indices(::Lagrange{3,RefPyramid,1}) = ((1,4,3,2), (1,2,5), (2,3,5), (3,4,5), (4,1,5), )
+edgedof_indices(::Lagrange{3,RefPyramid,1}) = ((1,2), (2,3), (3,4), (4,1), (1,5), (2,6), (3,5), (4,6))
+
+function reference_coordinates(::Lagrange{3,RefPyramid,1})
+    return [Vec{3, Float64}((0.0, 0.0, 0.0)),
+            Vec{3, Float64}((1.0, 0.0, 0.0)),
+            Vec{3, Float64}((1.0, 1.0, 0.0)),
+            Vec{3, Float64}((0.0, 1.0, .0)),
+            Vec{3, Float64}((0.0, 0.0, 1.0))]
+end
+
+function value(ip::Lagrange{3,RefPyramid,1}, i::Int, ξ::Vec{3})
+    (x,y,z) = ξ
+    zzero = z ≈ 1.0
+    i == 1 && return zzero ? 0.0 : (-x*y+(z-1)*(-x-y-z+1))/(z-1)
+    i == 2 && return zzero ? 0.0 : x*(y+z-1)/(z-1)
+    i == 3 && return zzero ? 0.0 : -x*y/(z-1)
+    i == 4 && return zzero ? 0.0 : y*(x+z-1)/(z-1)
+    i == 5 && return zzero ? 1.0 : z
```

Comment:
This one is at least not problematic:
```suggestion
    i == 5 && return z
```

For the other ones, if we are sufficiently close to z = 1, can we replac them with expressions that have the same value and derivative in the limit perhaps?

Reply:
There is no derivative at and close to z=1 with x=y=0. See e.g. l'Hospital on the basis function `i=3` with the coordinates x and y depending on z gives you $-(x'*y + y'*x)/(z'-1) = 0/0$ again. Maybe we should throw an error in this case for the derivative?